roboteam_ai/src/stp/plays/defensive/DefendPass.cpp

```diff
@@ -115,15 +119,18 @@ void DefendPass::calculateInfoForDefenders() noexcept {
         enemyMap.insert({score, enemy->getPos()});
     }
 
-    for (int i = 1; i <= 3; i++) {
-        if (!enemyMap.empty()) {
-            stpInfos["midfielder_" + std::to_string(i)].setPositionToDefend(enemyMap.rbegin()->second);
-            stpInfos["midfielder_" + std::to_string(i)].setBlockDistance(BlockDistance::HALFWAY);
-            enemyMap.erase(prev(enemyMap.end()));
-        } else {
-            break;
-        }
+    if (enemyMap.size() < 2) {
```

Comment:
What happens if there is exactly one enemy? Is that enemy then not defended? Or is that enemy by default then the keeper and does not need defending? I'm not sure having a keeper is required

Reply:
If there is one enemy they only have one robot that has a chance at scoring (enemy keeper is not included in this map). In that case our keeper should be enough to block the ball. A keeper is not required but its ID is set by the referee and can only be changed through the referee.

Reply:
there are some defenders to block the ball indeed